src/nvim/main.c

```diff
@@ -572,7 +593,18 @@ int main(int argc, char **argv)
   /*
    * Call the main command loop.  This never returns.
    */
-  normal_enter(false, false);
+  if (!is_remote_client) {
+    normal_enter(false, false);
+  } else {  
+    // Since a nvim client can have just one TUI,
```

Comment:
this entire else branch should be a function inside `tui.c` (i e the new tui_main).